### DISCUSSION TITLE
Closed or deleted projects still appear in Workspace Repository?

### DISCUSSION THREAD

**Discussion by btsweber** [eclipse-m2e/m2e-core#1266] [Q&A]
I have a workspace with two projects. Project B has a dependency on Project A. If I close project A and do Maven->Update Project... on Project B, Project B's configuration still references the Project A artifacts in my workspace rather than from my local repository. This is true even if I delete project B from my workspace and do Maven->Update Project... on project B. I notice that the Maven Repositories view continues to show Project A under Workspace Projects throughout.

I would expect that once I close Project A in my workspace it would no longer be eligible for workspace resolution, but that doesn't appear to be the case.

I'm using Eclipse 2020-06 and M2E 1.16

**Comment by mickaelistria** [comment]
Please first try latest release of Eclipse IDE and m2e. You usually won't get support for older releases.

**Reply by btsweber** [reply]
With Eclipse 2022-12 and M2E 2.1.2 I can confirm the following behaviors:

Given Project B with a dependency on Project A.

- If I close project A, and do Maven->Update Project... on Project B, configuration still references Project A in the workspace. I would expect it to reference project A in my local repo instead.
- If I restart Eclipse, Project B configuration still references project A in the workspace until I do Maven->Update Project... on project B, at which point configuration flips over to the local repository as expected.
- Open project A and then Maven->Update Project... on both Project A and Project B. Project B still references Project A in the local repository, not the workspace as expecrted.
- Restart Eclipse, Project B configuration now references project A in the workspace, as expected. No Update Project... required this time.

Seems like dependency resolution might not be listening to project open/close.

In Eclipse 2022-12, Maven Repositories view does not seem to function (local and workspace repository is grayed out) so no help there. I find that view quite helpful at times, so it's unfortunate that it seems broken now.

**Comment by mickaelistria** [comment]
Ok thanks, thanks for the details. I've created https://github.com/eclipse-m2e/m2e-core/issues/1284 for it. Let's continue discussing it there.
